test/api/v3/unit/libs/payments.test.js

```diff
@@ -17,38 +53,125 @@ describe('payments/index', () => {
+        recipient.purchased.plan = plan;
+        expect(recipient.purchased.plan.extraMonths).to.eql(0);
+        await api.createSubscription(data);
+        expect(recipient.purchased.plan.extraMonths).to.eql(3);
+      });
 
-      it('sets a dateTerminated date for a user without an existing subscription');
+      it('udpates date terminated for an existing plan with a terminated date', async () => {
+        let dateTerminated = new Date();
+        recipient.purchased.plan = plan;
+        recipient.purchased.plan.dateTerminated = dateTerminated;
 
-      it('sets plan.dateUpdated if it did not previously exist');
+        await api.createSubscription(data);
+        expect(recipient.purchased.plan.dateTerminated).to.eql( moment(dateTerminated).add(3, 'months').toDate());
+      });
 
-      it('does not change plan.customerId if it already exists');
+      it('sets a dateTerminated date for a user without an existing subscription', async () => {
+        expect(recipient.purchased.plan.dateTerminated).to.not.exist;
 
-      it('sets plan.customerId to "Gift" if it does not already exist');
+        await api.createSubscription(data);
 
-      it('increases the buyer\'s transaction count');
+        expect(recipient.purchased.plan.dateTerminated).to.exist;
+      });
 
-      it('sends a private message about the gift');
+      it('sets plan.dateUpdated if it did not previously exist', async () => {
+        expect(recipient.purchased.plan.dateUpdated).to.not.exist;
 
-      it('sends an email about the gift');
+        await api.createSubscription(data);
+
+        expect(recipient.purchased.plan.dateUpdated).to.exist;
+      });
 
-      it('sends a push notification about the gift');
+      it('does not change plan.customerId if it already exists', async () => {
+        recipient.purchased.plan = plan;
+        data.customerId = 'purchaserCustomerId';
+
+        expect(recipient.purchased.plan.customerId).to.eql('customer-id');
+
+        await api.createSubscription(data);
+
+        expect(recipient.purchased.plan.customerId).to.eql('customer-id');
+      });
+
+      it('sets plan.customerId to "Gift" if it does not already exist', async () => {
+        expect(recipient.purchased.plan.customerId).to.not.exist;
+
+        await api.createSubscription(data);
+
+        expect(recipient.purchased.plan.customerId).to.eql('Gift');
+      });
+
+      it('increases the buyer\'s transaction count', async () => {
+        expect(user.purchased.txnCount).to.eql(0);
+
+        await api.createSubscription(data);
 
-      it('tracks subscription purchase as gift (if prod)');
+        expect(user.purchased.txnCount).to.eql(1);
+      });
+
+      it('sends a private message about the gift', async () => {
+        await api.createSubscription(data);
+        expect(user.sendMessage).to.be.calledOnce;
+        expect(user.sendMessage).to.be.calledWith(recipient, `\`Hello recipient, sender has sent you 3 months of subscription!\``);
```

Comment:
To check that the message is properly localized is not really necessary as we test the underlying translation function. But instead of testing for the exact string you should use `t('stringKey')` so it won't be necessary to change the test when we change the string's content